server/src/internalClusterTest/java/org/elasticsearch/reservedstate/service/ComponentTemplatesFileSettingsIT.java

```diff
@@ -399,15 +399,17 @@ private void assertClusterStateSaveOK(CountDownLatch savedClusterState, AtomicLo
         boolean awaitSuccessful = savedClusterState.await(20, TimeUnit.SECONDS);
         assertTrue(awaitSuccessful);
 
-        final var response = client().execute(
-            GetComposableIndexTemplateAction.INSTANCE,
-            new GetComposableIndexTemplateAction.Request("template*")
-        ).get();
-
-        assertThat(
-            response.indexTemplates().keySet().stream().collect(Collectors.toSet()),
-            containsInAnyOrder("template_1", "template_2", "template_other")
-        );
+        assertBusy(() -> {
+            final var response = client().execute(
+                GetComposableIndexTemplateAction.INSTANCE,
+                new GetComposableIndexTemplateAction.Request("template*")
+            ).get();
+
+            assertThat(
+                response.indexTemplates().keySet().stream().collect(Collectors.toSet()),
+                containsInAnyOrder("template_1", "template_2", "template_other")
+            );
+        }, 60L, TimeUnit.SECONDS);
```

Comment:
From direct conversation: Can we wait for the appropriate metadata version?  The issue, as ya mentioned in conversation, is `GetComposableIndexTemplate` may be stale.